server/src/main/java/io/crate/analyze/TableParameters.java

```diff
@@ -67,7 +67,8 @@
 public class TableParameters {
 
     // all available table settings
-    static final NumberOfReplicasSetting NUMBER_OF_REPLICAS = new NumberOfReplicasSetting();
+    static final Map.Entry<String, NumberOfReplicasSetting> NUMBER_OF_REPLICAS =
+        Map.entry(stripIndexPrefix(NumberOfReplicasSetting.NAME), new NumberOfReplicasSetting());
```

Comment:
Trying to have the name accessible so that one would use this instead of `setting.getKey()` (which contains the prefix `index.`).

Reply:
to my understanding, `TableParameters` contains all settings available for `CREATE TABLE WITH/ ALTER TABLE SET`  and there quite some `index` prefixed settings, for instance,  `IndexMetadata.INDEX_BLOCKS_READ_SETTING` and so on.

I wonder what happens if we just use directly `IndexMetadata.SETTING_NUMBER_OF_REPLICAS `here instead of `NumberOfReplicasSetting`?

Reply:
Another question:
this setting is used [here](https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/analyze/MetadataToASTNodeResolver.java#L281) as well, afaik it's `SHOW CREATE TABLE` - could you please check diff what SHOW displays before/after the fix?

Reply:
`IndexMetadata.INDEX_BLOCKS_READ_SETTING` is grouped into `SUPPORTED_SETTINGS` and its `index.` prefix should have been already stripped before usage.

What is special about `NUMBER_OF_REPLICAS` is that it is directly used (without index stripping), for example, to create `CREATE_BLOB_TABLE_PARAMETERS` or `ALTER_BLOB_TABLE_PARAMETERS`, `NUMBER_OF_REPLICAS.getKey()` is called which should be wrong.

> I wonder what happens if we just use directly IndexMetadata.SETTING_NUMBER_OF_REPLICAS here instead of NumberOfReplicasSetting?

Currently, `NumberOfReplicasSetting.NAME == IndexMetadata.SETTING_NUMBER_OF_REPLICAS` so that is almost what is happening. If you would totally replace the field, I am not sure where would the instance of `NumberOfReplicasSetting` should go.

Reply:
> this setting is used [here](https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/analyze/MetadataToASTNodeResolver.java#L281) as well, afaik it's SHOW CREATE TABLE - could you please check diff what SHOW displays before/after the fix?

```
|    number_of_replicas = '1',                        |
```
still prints as `number_of_replicas` before and after the fix. We also have several tests for it in https://github.com/crate/crate/blob/master/server/src/test/java/io/crate/analyze/MetadataToASTNodeResolverTest.java